src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/api/TasksController.java

```diff
@@ -145,6 +149,31 @@ BuilderInfo scheduleTaskTesting(@PathParam("taskId") String taskId)
   BuilderInfo pollTaskTestingStatus(@PathParam("taskId") String taskId)
       throws TaskNotFoundException;
 
+  @GET
+  @Path("/{taskId}/files")
+  @ApiOperation(
+      value = "List all the files in a task skeleton",
+      response = String.class,
+      responseContainer = "List")
+  List<String> listFiles(
+      @PathParam("taskId") String taskId,
+      @QueryParam("usingTestingVersion") Boolean usingTestingVersion,
+      @QueryParam("variant") String variant)
+      throws TaskNotFoundException, TaskMissingVariantException, TaskStorageException;
+
+  @GET
+  @Path("/{taskId}/{files}/{fileName:.+}")
```

Comment:
Lets use /{taskId}/{variant}/skeleton/{fileName:.+}

(and lose the brackets round files)

Reply:
you have curly brackets round {files} which you don't want